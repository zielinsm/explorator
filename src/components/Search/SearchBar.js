import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { Icon, Input } from "semantic-ui-react";

class SearchBar extends PureComponent {
  constructor() {
    super();
    this.state = {
      value: ""
    };
    this.onKeyPress = this.handleKeyPress.bind(this);
    this.onInputChange = this.handleChange.bind(this);
    this.performSearch = this.goToSearch.bind(this);
  }

  handleChange(e) {
    this.setState({ value: e.target.value });
  }

  handleKeyPress(e) {
    if (e.key === "Enter") {
      this.goToSearch();
    }
  }

  goToSearch() {
    if (this.state.value.length > 0) {
      this.context.router.history.push("/results/" + this.state.value);
    }
  }

  render() {
    return (
      <div className="SearchBar">
        <Input
          icon={
            <Icon
              name="search"
              inverted
              circular
              link
              onClick={this.performSearch}
            />
          }
          placeholder="Search..."
          fluid
          onChange={this.onInputChange}
          onKeyPress={this.onKeyPress}
        />
      </div>
    );
  }

  static contextTypes = {
    router: PropTypes.object
  };
}

export default SearchBar;
