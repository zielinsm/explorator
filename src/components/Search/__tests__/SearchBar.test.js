import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import SearchBar from "../SearchBar";

describe("SearchBar component:", () => {
  let wrapper;
  const component = <SearchBar />;

  beforeEach(() => {
    wrapper = mount(component);
  });

  it("handles changes to the input", () => {
    wrapper.find("input").simulate("change", {
      target: { value: "hello" }
    });
    expect(wrapper.state("value")).toEqual("hello");
  });

  it("snapshot matches", () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
