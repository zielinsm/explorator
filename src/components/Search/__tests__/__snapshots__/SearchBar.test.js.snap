// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component: snapshot matches 1`] = `
<div
  className="SearchBar"
>
  <div
    className="ui fluid icon input"
  >
    <input
      disabled={undefined}
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Search..."
      tabIndex={undefined}
      type="text"
    />
    <i
      aria-hidden="true"
      className="search circular inverted link icon"
      onClick={[Function]}
    />
  </div>
</div>
`;
